src/test/components/ServiceWorkerManager.test.js

```diff
@@ -91,69 +106,178 @@ describe('app/ServiceWorkerManager', () => {
+
+      installOptions.onUpdateReady();
+      expect(serviceWorkerRuntime.applyUpdate).not.toHaveBeenCalled();
+      expect(container.firstChild).toBe(null);
+
+      // Some other tabs updated the SW.
+      installOptions.onUpdated();
+      expect(
+        ensureExists(container.querySelector('.photon-message-bar')).className
+      ).toMatch(/\bphoton-message-bar-warning\b/);
+      expect(container.firstChild).toMatchSnapshot();
+    });
   });
 
-  it('does not show a notice with the `from-addon` datasource', () => {
-    process.env.NODE_ENV = 'production';
+  describe('with the `from-addon` datasource', () => {
+    it(`doesn't show a notice if updated after we were fully loaded`, async () => {
+      process.env.NODE_ENV = 'production';
+
+      const { navigateToAddonLoadingPage, container, dispatch } = setup();
+      navigateToAddonLoadingPage();
+      await dispatch(viewProfile(_getSimpleProfile()));
+
+      const installOptions = serviceWorkerRuntime.install.mock.calls[0][0];
+
+      // We don't display anything if there's an update ready.
+      installOptions.onUpdateReady();
+      expect(serviceWorkerRuntime.applyUpdate).not.toHaveBeenCalled();
+      expect(container.firstChild).toBe(null);
+
+      // And we still don't if it was updated elsewhere.
+      installOptions.onUpdated();
+      expect(serviceWorkerRuntime.applyUpdate).not.toHaveBeenCalled();
+      expect(container.firstChild).toBe(null);
+    });
+
+    it('shows a warning notice if updated before we were ready', async () => {
+      process.env.NODE_ENV = 'production';
+
+      const { navigateToAddonLoadingPage, container, dispatch } = setup();
+      navigateToAddonLoadingPage();
+
+      const installOptions = serviceWorkerRuntime.install.mock.calls[0][0];
 
-    const { navigateToAddonLoadingPage, container } = setup();
-    navigateToAddonLoadingPage();
+      // We don't display anything if there's an update ready.
+      installOptions.onUpdateReady();
+      expect(serviceWorkerRuntime.applyUpdate).not.toHaveBeenCalled();
+      expect(container.firstChild).toBe(null);
 
-    const installOptions = serviceWorkerRuntime.install.mock.calls[0][0];
-    installOptions.onUpdated();
-    expect(container.firstChild).toBe(null);
+      // But we do if it's updated from another tab.
+      installOptions.onUpdated();
+      expect(serviceWorkerRuntime.applyUpdate).not.toHaveBeenCalled();
+      expect(
+        ensureExists(container.querySelector('.photon-message-bar')).className
+      ).toMatch(/\bphoton-message-bar-warning\b/);
+      expect(container.firstChild).toMatchSnapshot();
+
+      // The notice stays if we're getting ready after that.
+      await dispatch(viewProfile(_getSimpleProfile()));
+      expect(container.firstChild).not.toBe(null);
+    });
   });
 
   describe('automatic reloading', () => {
     it('reloads the application automatically if there is an error and there is a pending version', () => {
```

Comment:
note: this doesn't work properly at the moment, because we lose the hash in this situation. I'd like to look at this soon too...